dspace-spring-rest/src/test/java/org/dspace/app/opensearch/OpenSearchControllerDisabledTest.java

```diff
@@ -0,0 +1,200 @@
+
+    private ConfigurationService configurationService;
+
+    @Before
+    public void init() throws Exception {
+        configurationService = DSpaceServicesFactory.getInstance().getConfigurationService();
+        configurationService.setProperty("websvc.opensearch.enable", false);
+/*
+System.out.println("Testing OpenSearch");
+MockHttpServletResponse resp2 = getClient().perform(get("/opensearch/search")
+                                .param("query", "dog"))
+                               .andReturn().getResponse();
+System.out.println("Response from Test 2: "+resp2.getContentAsString());
+*/
+    }
+
+    @Test
+    public void searchTest() throws Exception {
+        //When we call the root endpoint
+        getClient().perform(get("/opensearch/search")
+                                .param("query", "dog"))
+                   //The status has to be 200 OK
+                   .andExpect(status().isOk())
+                   //We expect the content type to be "application/hal+json;charset=UTF-8"
+                   .andExpect(content().contentType("text/html"))
+/*
+                   //Our default Discovery config has 4 browse indexes so we expect this to be reflected in the page
+                   // object
+                   .andExpect(jsonPath("$.page.size", is(20)))
+                   .andExpect(jsonPath("$.page.totalElements", is(4)))
+                   .andExpect(jsonPath("$.page.totalPages", is(1)))
+                   .andExpect(jsonPath("$.page.number", is(0)))
+
+                   //The array of browse index should have a size 4
+                   .andExpect(jsonPath("$._embedded.browses", hasSize(4)))
+
+                   //Check that all (and only) the default browse indexes are present
+                   .andExpect(jsonPath("$._embedded.browses", containsInAnyOrder(
+                       BrowseIndexMatcher.dateIssuedBrowseIndex("asc"),
+                       BrowseIndexMatcher.contributorBrowseIndex("asc"),
+                       BrowseIndexMatcher.titleBrowseIndex("asc"),
+                       BrowseIndexMatcher.subjectBrowseIndex("asc")
+                   )))
+*/
+        ;
+    }
+
+    @Test
+    public void serviceDocumentTest() throws Exception {
+        //When we call the root endpoint
+        getClient().perform(get("/opensearch/service"))
+                   //The status has to be 200 OK
+                   .andExpect(status().isOk())
+                   .andExpect(content().contentType("text/html"))
+                   //We expect the content type to be "application/hal+json;charset=UTF-8"
+//                   .andExpect(content().contentType(contentType))
+/*
+                   //Check that the JSON root matches the expected browse index
+                   .andExpect(jsonPath("$", BrowseIndexMatcher.titleBrowseIndex("asc")))
+*/
+        ;
+    }
+
+/*
+    @Test
+    public void findItemsWithOU() throws Exception {
```

Comment:
Same here (as above). More commented out tests that could use an explanation.

Reply:
removed comments